app/src/main/java/com/nutomic/syncthingandroid/service/RestApi.java

```diff
@@ -390,46 +404,14 @@ public void getConnections(final OnResultListener1<Connections> listener) {
         });
     }
 
-    /**
-     * Calculates completion percentage for the given device using {@link #mCachedModelInfo}.
-     */
-    private int getDeviceCompletion(String deviceId) {
-        int folderCount = 0;
-        float percentageSum = 0;
-        // Syncthing UI limits pending deletes to 95% completion of a device
-        int maxPercentage = 100;
-        for (Map.Entry<String, Model> modelInfo : mCachedModelInfo.entrySet()) {
-            boolean isShared = false;
-            for (Folder r : getFolders()) {
-                if (r.getDevice(deviceId) != null) {
-                    isShared = true;
-                    break;
-                }
-            }
-            if (isShared) {
-                long global = modelInfo.getValue().globalBytes;
-                long local = modelInfo.getValue().inSyncBytes;
-                if (modelInfo.getValue().needFiles == 0 && modelInfo.getValue().needDeletes > 0)
-                    maxPercentage = 95;
-                percentageSum += (global != 0)
-                        ? (local * 100f) / global
-                        : 100f;
-                folderCount++;
-            }
-        }
-        return (folderCount != 0)
-                ? Math.min(Math.round(percentageSum / folderCount), maxPercentage)
-                : 100;
-    }
-
     /**
      * Returns status information about the folder with the given id.
      */
-    public void getModel(final String folderId, final OnResultListener2<String, Model> listener) {
-        new GetRequest(mContext, mUrl, GetRequest.URI_MODEL, mApiKey,
+    public void getFolderStatus(final String folderId, final OnResultListener2<String, FolderStatus> listener) {
+        new GetRequest(mContext, mUrl, GetRequest.URI_STATUS, mApiKey,
                     ImmutableMap.of("folder", folderId), result -> {
-            Model m = new Gson().fromJson(result, Model.class);
-            mCachedModelInfo.put(folderId, m);
+            FolderStatus m = new Gson().fromJson(result, FolderStatus.class);
+            mCachedFolderStatusInfo.put(folderId, m);
```

Comment:
Why can't you just pass m? I assume folderId is an attribute on that struct.

Reply:
No, it isn't. See FolderStatus.java. Anyways, I'd like the idea to put that attribute in, I think that wouldn't help us simplifying things because that "old" code before the PR needs folderId as a key on the HashMap.